podcast/2.10.md

```diff
@@ -0,0 +1,778 @@
+
+**Beyang Liu:**
+
+Makes sense. I think that's an important lesson that every, I won't say every, but probably a lot of engineers discover over time–migrations are super difficult. So the Piper team, essentially hired, it sounds like, their closest competitor, and that was the end of it.
+
+**Han-Wen Nienhuys:**
+
+You could see it that way. So the Piper team was in Munich and I relocated to Munich. I'm still in Munich–if I look outside I see the beautiful blue German sky.
+
+**Beyang Liu:**
+
+Awesome.
+
+**Han-Wen Nienhuys:**
+
+And so I worked on the Piper team for a while. So the Git wrapper was a 20% project–a classic 20% project. What is quite special is for you to create a 20% project and then leave it with another team, but I managed to do that. At some point I decided to work on another developer tool, so I worked on open sourcing our build system. And when I switched teams, I left this Git wrapper with the Piper team.
+
+**Beyang Liu:**
+
+Got it.
+
+**Han-Wen Nienhuys:**
+
+And they deleted it. So they built a replacement based on Mercurial–similar to Facebook, I think.
+
+**Beyang Liu:**
+
+Yeah, yeah.
+
+**Han-Wen Nienhuys:**
+
+And so that is now the de facto way to do distributed workflows at Google.
+
+**Beyang Liu:**
+
+So most Google engineers today see a Mercurial-like interface?
+
+**Han-Wen Nienhuys:**
+
+I'm not keeping track of how many people there are but if you want to do... So one of the things that is hard with Perforce is that, let's say you want to implement a feature but the feature needs a bug fix.
+
+**Beyang Liu:**
+
+Mm-hmm.
+
+**Han-Wen Nienhuys:**
+
+And so what you would do in Git is that first you make a commit that fixes the bug and then you do a commit on top of that that does the feature, and then maybe there's review and then, depending on what workflow you do, you can actually evolve both the bug fix and the feature independently.
+
+With Perforce that's really difficult to do. You can kind of do it if you make sure that the fix, the bug fix, and the feature don't touch the same files but if it touches the same files then you're basically screwed.
+
+**Han-Wen Nienhuys:**
+
+And so that is the workflow that my Git wrapper made possible and that's also the workflow that this next generation of that ,which is based on Mercurial, made possible.
+
+**Beyang Liu:**
+
+Got it. I'm just curious because I noticed in open source you've done some work on FUSEs, Filesystem in USErspace. Is that how you initially got into FUSEs, this work on Piper or did that come later?
+
+**Han-Wen Nienhuys:**
+
+No, it actually came earlier. Part of the reason why this Git Wrapper was working was that Google had a file system, a virtual file system that gave you all the files in the repository available locally, and of course they're not really available because they're only once you actually go into the directory and try to read the file then the file system would go and fetch the file.
```

Comment:
```suggestion
No, it actually came earlier. Part of the reason why this Git wrapper was working was that Google had a file system, a virtual file system that gave you all the files in the repository available locally, and of course they're not really available because they're only once you actually go into the directory and try to read the file then the file system would go and fetch the file.
```